dace/transformation/dataflow/redundant_array.py

```diff
@@ -187,15 +197,59 @@ def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
         # Make sure that the candidate is a transient variable
         if not in_desc.transient:
             return False
+        
+        # 1. Get edge e1 and extract subsets for arrays A and B
+        e1 = graph.edges_between(in_array, out_array)[0]
+        a1_subset, b_subset = _validate_subsets(e1, sdfg.arrays)
 
         if strict:
             # In strict mode, make sure the memlet covers the removed array
-            edge = graph.edges_between(in_array, out_array)[0]
-            subset = copy.deepcopy(edge.data.subset)
+            subset = copy.deepcopy(e1.data.subset)
             subset.squeeze()
             shape = [sz for sz in in_desc.shape if sz != 1]
             if any(m != a for m, a in zip(subset.size(), shape)):
                 return False
+
+            # NOTE: Library node check
```

Comment:
with this code, can we remove the reduce check below?

Reply:
The reduce check is for a different issue. It is related to reductions with the `keepdims` keyword enabled. To generate the correct SDFG, we first generate the reduction ignoring `keepdims` and then we copy to the final output. We are not checking any relation between the input and the output. Also the new library node check does not ensure that the "intermediate" reduction output in the case of `keepdims=True` is not removed.

Reply:
Let me also clarify that the problem with `keepdims=True` is that our Reduce node expansions are not compatible with writing directly to the final keepdims-enabled output. And we have to many reduce expansions to fix them now.

Reply:
ok